### PR TITLE
[BUGFIX beta] property_set should reuse reference to its meta

### PR DESCRIPTION
**Pull request by stefanpenner**

### CODE REVIEW

**Comment by scalvert** [comment]
Good catch! No peeking :)

**Review by krisselden — APPROVED** [review]

File: packages/ember-runtime/lib/mixins/array.js [hunk 1]
```diff
@@ -653,22 +653,26 @@ EachProxy.prototype = {
   arrayWillChange(content, idx, removedCnt, addedCnt) {
     let keys = this._keys;
     let lim = removedCnt > 0 ? idx + removedCnt : -1;
+    let meta;
     for (let key in keys) {
+      meta = meta || peakMeta(this);
```
Comment (runspired) [inline packages/ember-runtime/lib/mixins/array.js]: is this a misspelling ?

Comment (runspired) [inline packages/ember-runtime/lib/mixins/array.js]: It is, should be `peekMeta`

Comment (runspired) [inline packages/ember-runtime/lib/mixins/array.js]: although I do like that we've hit "peak meta"

File: packages/ember-runtime/lib/mixins/array.js [hunk 2]
```diff
@@ -653,22 +653,26 @@ EachProxy.prototype = {
...
+      propertyWillChange(this, key, meta);
     }
   },
 
   arrayDidChange(content, idx, removedCnt, addedCnt) {
     let keys = this._keys;
     let lim = addedCnt > 0 ? idx + addedCnt : -1;
+    let meta;
     for (let key in keys) {
+      meta = meta || peakMeta(this);
```
Comment (runspired) [inline packages/ember-runtime/lib/mixins/array.js]: same as above, `peekMeta`

**Review by runspired — CHANGES_REQUESTED** [review]
Fix da peak

**Review by runspired — APPROVED** [review]

**Comment by stefanpenner** [comment]
@bmeurer this should eat into the 5% of time spent in weakmap lookup you guys where seeing.

**Comment by stefanpenner** [comment]
@homu r+

**Comment by stefanpenner** [comment]
@homu retry

**Comment by stefanpenner** [comment]
@homu what's taking you so long :(

**Comment by rwjblue** [comment]
I think homu is basically broken here :(

**Comment by bmeurer** [comment]
I don't have the full context on this change.

**Comment by stefanpenner** [comment]
Sorry. This pull request removes repeated and unnecessary `weakmap get`'s... I realize you guys are also working on improving weakmap perf so I figure I would loop you in.

TL;DR this pr reduces the number of WeakMapGets's to at least 1/3 in some cases.

**Comment by bmeurer** [comment]
Ah I see. That makes total sense, no matter what V8 (or any other engine) is doing.

**Comment by stefanpenner** [comment]
@bmeurer yup, should work in combination with any work engines do.
